src/Orchard.Environment.Shell.Data/ShellDescriptorManager.cs

```diff
@@ -31,14 +32,20 @@ public class ShellDescriptorManager : IShellDescriptorManager
             _eventBus = eventBus;
             _session = session;
             _logger = logger;
+
+            _asyncLazyShellDescriptor = new Lazy<Task<ShellDescriptor>>(() => Task.Run(() =>
```

Comment:
I agree, it was to improve time because of other things (see my other comments) but found a better and simpler way.